### PR TITLE
[BOP-775] RBAC migration

### PR DESCRIPTION
**Pull request by nwneisen**
https://mirantis.jira.com/browse/BOP-775

Adds information about RBAC to the migration section

### CODE REVIEW

**Comment by nwneisen** [comment]
@KoryKessel-Mirantis I started creating this. I'm not a fan of the formatting so I figured I'd draft it and see what you think

File: .gitignore [hunk 1]
```diff
@@ -15,4 +15,4 @@
 **/mke4.yaml
 **/mke.yaml
 
-./public/*
+public/*
```
Comment (nwneisen) [inline .gitignore:18]: Typo to ignore the public folder. Fixed with this change

File: content/docs/migrate-from-MKE-3.md [hunk 2]
```diff
@@ -142,3 +142,156 @@ client bundle. The docker swarm cluster will no longer be accessible as well.
...
+MKE 4 uses a different RBAC model than MKE 3. MKE 4 no longer supports swarm mode
+and uses Kubernetes RBAC for authorization. As a result, the RBAC configuration
+for Swarm mode does not exist in MKE 4. However, the Swarm mode roles have always
+had Kubernetes RBAC roles associated with them. The upgrade process migrates these
+Kubernetes roles and bindings to the new MKE 4 cluster.
+
+The following sections list the MKE 3 Swarm mode roles and their corresponding
+Kubernetes RBAC permissions:
+
+### View Only
```
Comment (KoryKessel-Mirantis) [inline content/docs/migrate-from-MKE-3.md:157]: I suggest that the #### heading level be a dropdown heading that expands when clicked to reveal the bullet detail.

I found information on how to do this in markdown here: https://gist.github.com/citrusui/07978f14b11adada364ff901e27c7f61

Thus, the ### headings remain, the #### headings become the text that contains the dropdown text, and the dropdown text will be the bullet lists.

File: content/docs/migrate-from-MKE-3.md [hunk 3]
```diff
@@ -142,3 +142,156 @@ client bundle. The docker swarm cluster will no longer be accessible as well.
...
+
+## RBAC Migrations
+
+MKE 4 uses a different RBAC model than MKE 3. MKE 4 no longer supports swarm mode
+and uses Kubernetes RBAC for authorization. As a result, the RBAC configuration
+for Swarm mode does not exist in MKE 4. However, the Swarm mode roles have always
+had Kubernetes RBAC roles associated with them. The upgrade process migrates these
+Kubernetes roles and bindings to the new MKE 4 cluster.
+
+The following sections list the MKE 3 Swarm mode roles and their corresponding
```
Comment (KoryKessel-Mirantis) [inline content/docs/migrate-from-MKE-3.md:154]:
```suggestion
The sections that follow detail the MKE 3 Swarm mode roles and their corresponding
```

File: content/docs/migrate-from-MKE-3.md [hunk 4]
```diff
@@ -142,3 +142,156 @@ client bundle. The docker swarm cluster will no longer be accessible as well.
...
 In the event of an upgrade failure, the upgrade process rolls back,
 restoring the MKE 3 cluster to its original state.
+
+## RBAC Migrations
+
+MKE 4 uses a different RBAC model than MKE 3. MKE 4 no longer supports swarm mode
+and uses Kubernetes RBAC for authorization. As a result, the RBAC configuration
+for Swarm mode does not exist in MKE 4. However, the Swarm mode roles have always
+had Kubernetes RBAC roles associated with them. The upgrade process migrates these
+Kubernetes roles and bindings to the new MKE 4 cluster.
```
Comment (KoryKessel-Mirantis) [inline content/docs/migrate-from-MKE-3.md:152]:
```suggestion
The RBAC model MKE 4 uses differs significantly from the one in use by MKE 3.

As MKE 4 no longer supports Swarm mode, the platform uses Kubernetes RBAC for authorization. Thus, the RBAC configuration for Swarm mode does not exist in MKE 4. The Swarm mode roles have Kubernetes RBAC roles associated with them, however, and the MKE 4 upgrade process migrates these
Kubernetes roles and bindings to the new MKE 4 cluster.
```

File: content/docs/migrate-from-MKE-3.md [hunk 5]
```diff
@@ -142,3 +142,156 @@ client bundle. The docker swarm cluster will no longer be accessible as well.
...
+- Service
+- Stack
+- StatefulSet
+- StorageClass
+- ThirdPartyResource
+- User
+
+### Scheduler
+
+No kubernetes RBAC permissions are associated with the Scheduler role.
```
Comment (KoryKessel-Mirantis) [inline content/docs/migrate-from-MKE-3.md:297]:
```suggestion
No Kubernetes RBAC permissions correlate to the Scheduler role.
```

File: content/docs/migrate-from-MKE-3.md (hunk 2 above)
Comment (nwneisen) [reply content/docs/migrate-from-MKE-3.md:157]: Done

**Review by KoryKessel-Mirantis — APPROVED** [review]

**Pull request merged**